### PR TITLE
fix scoping regression with calls in generic bodies

### PR DESCRIPTION
**Pull request by metagn**
refs #22029, refs https://github.com/status-im/nim-libp2p/actions/runs/5263850340/jobs/9514434659

### CODE REVIEW

File: compiler/semcall.nim [hunk 1]
```diff
@@ -628,7 +628,7 @@ proc semOverloadedCall(c: PContext, n, nOrig: PNode,
     result = semResolvedCall(c, r, n, flags)
   else:
     if efDetermineType in flags and c.inGenericContext > 0 and c.matchedConcept == nil:
-      result = n
+      result = semGenericStmt(c, n)
```
Comment (Araq) [inline compiler/semcall.nim:631]: What? Why?

**Comment by Araq** [comment]
Seems to go into the wrong direction. What commit caused this regression? What else can we do about it?

**Comment by metagn** [comment]
#22029 caused it, basically a `tyFromExpr` node that uses a private symbol and is instantiated from another module doesn't find that symbol. 

Used `semGenericStmt` because it solves this scoping issue by resolving the symbol, and is already used [here](https://github.com/nim-lang/Nim/blob/647d9611ae81c7af2c86af5b33a14787254896b5/compiler/semtypes.nim#L1295) in `semtypes` for generic proc default values and there's also [this use](https://github.com/nim-lang/Nim/blob/647d9611ae81c7af2c86af5b33a14787254896b5/compiler/semtypes.nim#L2030) which also calls `makeTypeFromExpr` but I don't really know what it does.

Don't really know another solution. I guess we could also *always* use `semGenericStmt` for `range`/`array`/`when` parameters in generic types but this is a radical change.

**Comment by Araq** [comment]
Fine.

**Pull request merged**